trello/trelloclient.py

```diff
@@ -212,13 +212,14 @@ def fetch_json(
             headers = {}
         if query_params is None:
             query_params = {}
-        if post_args is None:
-            post_args = {}
 
-        # if files specified, we don't want any data
+        # Per trello api specificacion payload should not be present on get requests.
         data = None
+        
+        # if files specified, we don't want any data
         if files is None:
-            data = json.dumps(post_args)
+            if post_args is not None:
```

Comment:
I would rather check if it's a get so you don't send body data on a get even if some has been specified, which could be achieved with the line below instead:
```
if post_args and http_method != 'GET'
```

Reply:
I gree

Reply:
Better:

```python
data = None if files is not None or method == 'GET' else json.dumps(post_args)
```

Data must have None if:
- Files has value (is not None)
- method is GET regardless data value (cannot post data or files using a GET)

With this method it is not necessary to remove the initial section of code:
```python
if post_args is None:
  post_args = {}
```